app/src/main/java/com/example/tasky/feature_agenda/presentation/agenda_screen/AgendaScreen.kt

```diff
@@ -0,0 +1,119 @@
+package com.example.tasky.feature_agenda.presentation.agenda_screen
+
+import android.widget.Toast
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.navigation.NavController
+import com.example.tasky.feature_agenda.presentation.agenda_screen.components.DatePickerDialog
+import com.example.tasky.feature_agenda.presentation.agenda_screen.components.DayChip
+import com.example.tasky.feature_agenda.presentation.agenda_screen.components.TopSection
+import com.example.tasky.presentation.theme.BackgroundBlack
+import com.example.tasky.presentation.theme.BackgroundWhite
+import com.example.tasky.util.Result
+import com.example.tasky.util.Screen
+import com.vanpra.composematerialdialogs.rememberMaterialDialogState
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+
+@Composable
+fun AgendaScreen(
+    navController: NavController,
+    viewModel: AgendaViewModel = hiltViewModel()
+) {
```

Comment:
I'd avoid passing the ViewModel directly to screen composables. This has 2 major reasons:
1. You're making it harder to have a preview, since you'll need to provide a fake ViewModel instance and fakes for everything the ViewModel needs in its constructor.
2. UI tests are also a bit harder since you'll need to provide these fakes.

What I suggest instead:
```
// In your ViewModel
@HiltViewModel
class SampleViewModel @Inject constructor(): ViewModel() {
    private val _state = MutableStateFlow(SampleState())
    val state = _state.asStateFlow()

    fun onEvent(event: SampleEvent) {
        when(event) {
            is SampleEvent.Event1 -> // Handle event1
            is SampleEvent.Event2 -> // Handle event2
            // ...
        }
    }

    // ...
}

// In your NavHost
composable(Routes.sample) {
    val viewModel = hiltViewModel<SampleViewModel>()
    val state by viewModel.state.collectAsStateWithLifecycle()
    SampleScreen(
        state = state,
        onEvent = viewModel::onEvent
    )
}

// Your screen composable
@Composable
fun SampleScreen(
    state: SampleState,
    onEvent: (SampleEvent) -> Unit
) {
    // ...
    Button(onClick = { onEvent(SampleEvent.ClickedButtonX) } {
        Text("Click me")
    }
}
```